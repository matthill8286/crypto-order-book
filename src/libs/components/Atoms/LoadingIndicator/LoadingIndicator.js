function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { rotateForward } from "../../../styles/animation";
import { styled } from "../../../styles/styled";
var StyledLoadingWrapper = styled.div.withConfig({
  displayName: "LoadingIndicator__StyledLoadingWrapper",
  componentId: "sc-12we8y6-0"
})(["width:", "px;height:", "px;display:", ";position:absolute;& div{box-sizing:border-box;display:block;position:absolute;width:100%;height:100%;border:", "px solid;border-radius:50%;animation:", " 1.2s cubic-bezier(0.5,0,0.5,1) infinite;border-color:", " transparent transparent transparent;}& div:nth-child(1){animation-delay:-0.45s;}& div:nth-child(2){animation-delay:-0.3s;}& div:nth-child(3){animation-delay:-0.15s;}"], function (_ref) {
  var size = _ref.size;
  return size;
}, function (_ref2) {
  var size = _ref2.size;
  return size;
}, function (_ref3) {
  var isVisible = _ref3.isVisible;
  return isVisible ? 'block' : 'none';
}, function (_ref4) {
  var size = _ref4.size;
  return size * 0.1;
}, rotateForward, function (_ref5) {
  var theme = _ref5.theme,
      color = _ref5.color;
  return theme.color[color];
});
export var StyledLoadingWrapperAlternate = styled.ul.withConfig({
  displayName: "LoadingIndicator__StyledLoadingWrapperAlternate",
  componentId: "sc-12we8y6-1"
})(["position:absolute;display:", ";margin:0 auto;padding:0;justify-content:center;height:", "px;li{list-style:none;width:", "px;height:", "px;margin:0 3px;animation:animate 0.4s infinite alternate;background:", ";&:nth-child(2){animation-delay:0.1s;}&:nth-child(3){animation-delay:0.2s;}}@keyframes animate{0%{transform:scaleY(1);}25%{transform:scaleY(1);}50%{transform:scaleY(1);}75%{transform:scaleY(1);}100%{transform:scaleY(1.7);}}"], function (_ref6) {
  var isVisible = _ref6.isVisible;
  return isVisible ? 'flex' : 'none';
}, function (_ref7) {
  var size = _ref7.size;
  return size * 0.3;
}, function (_ref8) {
  var barWidth = _ref8.barWidth;
  return barWidth * 0.05;
}, function (_ref9) {
  var size = _ref9.size;
  return size * 0.3;
}, function (_ref10) {
  var theme = _ref10.theme,
      color = _ref10.color;
  return theme.color[color];
});
export var LoadingIndicator = function LoadingIndicator(_ref11) {
  var isVisible = _ref11.isVisible,
      _ref11$size = _ref11.size,
      size = _ref11$size === void 0 ? 50 : _ref11$size,
      _ref11$barWidth = _ref11.barWidth,
      barWidth = _ref11$barWidth === void 0 ? 15 : _ref11$barWidth,
      _ref11$color = _ref11.color,
      color = _ref11$color === void 0 ? 'primary' : _ref11$color,
      otherProps = _objectWithoutProperties(_ref11, ["isVisible", "size", "barWidth", "color"]);

  return React.createElement(StyledLoadingWrapper, _extends({
    isVisible: isVisible,
    size: size,
    barWidth: barWidth,
    color: color
  }, otherProps), React.createElement("div", null), React.createElement("div", null), React.createElement("div", null), React.createElement("div", null));
};
//# sourceMappingURL=LoadingIndicator.js.map