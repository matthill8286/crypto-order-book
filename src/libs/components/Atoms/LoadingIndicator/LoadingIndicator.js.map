{"version":3,"sources":["../../../../src/components/Atoms/LoadingIndicator/LoadingIndicator.tsx"],"names":["React","rotateForward","styled","StyledLoadingWrapper","div","size","isVisible","theme","color","StyledLoadingWrapperAlternate","ul","barWidth","LoadingIndicator","otherProps"],"mappings":";;;;;;AAAA,OAAOA,KAAP,MAAkB,OAAlB;AACA,SAASC,aAAT;AACA,SAASC,MAAT;AAkBA,IAAMC,oBAAoB,GAAGD,MAAM,CAACE,GAAV;AAAA;AAAA;AAAA,0aACf;AAAA,MAAGC,IAAH,QAAGA,IAAH;AAAA,SAAcA,IAAd;AAAA,CADe,EAEd;AAAA,MAAGA,IAAH,SAAGA,IAAH;AAAA,SAAcA,IAAd;AAAA,CAFc,EAGb;AAAA,MAAGC,SAAH,SAAGA,SAAH;AAAA,SAAoBA,SAAS,GAAG,OAAH,GAAa,MAA1C;AAAA,CAHa,EAYZ;AAAA,MAAGD,IAAH,SAAGA,IAAH;AAAA,SAAcA,IAAI,GAAG,GAArB;AAAA,CAZY,EAcTJ,aAdS,EAeN;AAAA,MAAGM,KAAH,SAAGA,KAAH;AAAA,MAAUC,KAAV,SAAUA,KAAV;AAAA,SAAsBD,KAAK,CAACC,KAAN,CAAYA,KAAZ,CAAtB;AAAA,CAfM,CAA1B;AA+BA,OAAO,IAAMC,6BAA6B,GAAGP,MAAM,CAACQ,EAAV;AAAA;AAAA;AAAA,wbAE7B;AAAA,MAAGJ,SAAH,SAAGA,SAAH;AAAA,SAAoBA,SAAS,GAAG,MAAH,GAAY,MAAzC;AAAA,CAF6B,EAM9B;AAAA,MAAGD,IAAH,SAAGA,IAAH;AAAA,SAAcA,IAAI,GAAG,GAArB;AAAA,CAN8B,EAS7B;AAAA,MAAGM,QAAH,SAAGA,QAAH;AAAA,SAAkBA,QAAQ,GAAG,IAA7B;AAAA,CAT6B,EAU5B;AAAA,MAAGN,IAAH,SAAGA,IAAH;AAAA,SAAcA,IAAI,GAAG,GAArB;AAAA,CAV4B,EAaxB;AAAA,MAAGE,KAAH,UAAGA,KAAH;AAAA,MAAUC,KAAV,UAAUA,KAAV;AAAA,SAAsBD,KAAK,CAACC,KAAN,CAAYA,KAAZ,CAAtB;AAAA,CAbwB,CAAnC;AA0CP,OAAO,IAAMI,gBAAiD,GAAG,SAApDA,gBAAoD,SAM3D;AAAA,MALJN,SAKI,UALJA,SAKI;AAAA,2BAJJD,IAII;AAAA,MAJJA,IAII,4BAJG,EAIH;AAAA,+BAHJM,QAGI;AAAA,MAHJA,QAGI,gCAHO,EAGP;AAAA,4BAFJH,KAEI;AAAA,MAFJA,KAEI,6BAFI,SAEJ;AAAA,MADDK,UACC;;AACJ,SACE,oBAAC,oBAAD;AACE,IAAA,SAAS,EAAEP,SADb;AAEE,IAAA,IAAI,EAAED,IAFR;AAGE,IAAA,QAAQ,EAAEM,QAHZ;AAIE,IAAA,KAAK,EAAEH;AAJT,KAKMK,UALN,GAME,gCANF,EAOE,gCAPF,EAQE,gCARF,EASE,gCATF,CADF;AAaD,CApBM","sourcesContent":["import React from 'react'\nimport { rotateForward } from '@/styles/animation'\nimport { styled } from '@/styles/styled'\n\nexport type LoadingIndicatorColor = 'white' | 'black' | 'primary'\n\nexport interface StyledLoadingIndicatorProps {\n  isVisible: boolean\n  size: number\n  barWidth: number\n  color: LoadingIndicatorColor\n}\n\nexport interface LoadingIndicatorProps {\n  isVisible: boolean\n  size?: number\n  barWidth?: number\n  color?: LoadingIndicatorColor\n}\n\nconst StyledLoadingWrapper = styled.div<StyledLoadingIndicatorProps>`\n  width: ${({ size }) => size}px;\n  height: ${({ size }) => size}px;\n  display: ${({ isVisible }) => (isVisible ? 'block' : 'none')};\n  position: absolute;\n\n  & div {\n    box-sizing: border-box;\n    display: block;\n    position: absolute;\n    width: 100%;\n    height: 100%;\n    border: ${({ size }) => size * 0.1}px solid;\n    border-radius: 50%;\n    animation: ${rotateForward} 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;\n    border-color: ${({ theme, color }) => theme.color[color]} transparent transparent transparent;\n  }\n\n  & div:nth-child(1) {\n    animation-delay: -0.45s;\n  }\n\n  & div:nth-child(2) {\n    animation-delay: -0.3s;\n  }\n\n  & div:nth-child(3) {\n    animation-delay: -0.15s;\n  }\n`\n\nexport const StyledLoadingWrapperAlternate = styled.ul<StyledLoadingIndicatorProps>`\n  position: absolute;\n  display: ${({ isVisible }) => (isVisible ? 'flex' : 'none')};\n  margin: 0 auto;\n  padding: 0;\n  justify-content: center;\n  height: ${({ size }) => size * 0.3}px;\n  li {\n    list-style: none;\n    width: ${({ barWidth }) => barWidth * 0.05}px;\n    height: ${({ size }) => size * 0.3}px;\n    margin: 0 3px;\n    animation: animate 0.4s infinite alternate;\n    background: ${({ theme, color }) => theme.color[color]};\n\n    &:nth-child(2) {\n      animation-delay: 0.1s;\n    }\n    &:nth-child(3) {\n      animation-delay: 0.2s;\n    }\n  }\n\n  @keyframes animate {\n    0% {\n      transform: scaleY(1);\n    }\n    25% {\n      transform: scaleY(1);\n    }\n    50% {\n      transform: scaleY(1);\n    }\n    75% {\n      transform: scaleY(1);\n    }\n    100% {\n      transform: scaleY(1.7);\n    }\n  }\n`\n\nexport const LoadingIndicator: React.FC<LoadingIndicatorProps> = ({\n  isVisible,\n  size = 50,\n  barWidth = 15,\n  color = 'primary',\n  ...otherProps\n}) => {\n  return (\n    <StyledLoadingWrapper\n      isVisible={isVisible}\n      size={size}\n      barWidth={barWidth}\n      color={color}\n      {...otherProps}>\n      <div />\n      <div />\n      <div />\n      <div />\n    </StyledLoadingWrapper>\n  )\n}\n"],"file":"LoadingIndicator.js"}