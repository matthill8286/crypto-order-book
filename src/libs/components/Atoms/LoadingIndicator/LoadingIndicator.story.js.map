{"version":3,"sources":["../../../../src/components/Atoms/LoadingIndicator/LoadingIndicator.story.tsx"],"names":["number","radios","storiesOf","React","LoadingIndicator","loadingIndicatorColor","white","black","primary","knobs","isVisible","color","size","module","add"],"mappings":"AAAA,SAASA,MAAT,EAAiBC,MAAjB,QAA+B,wBAA/B;AACA,SAASC,SAAT,QAA0B,kBAA1B;AACA,OAAO,KAAKC,KAAZ,MAAuB,OAAvB;AACA,SAASC,gBAAT;AAEA,IAAMC,qBAA+D,GAAG;AACtEC,EAAAA,KAAK,EAAE,OAD+D;AAEtEC,EAAAA,KAAK,EAAE,OAF+D;AAGtEC,EAAAA,OAAO,EAAE;AAH6D,CAAxE;;AAMA,IAAMC,KAAK,GAAG,SAARA,KAAQ,GAA6B;AACzC,SAAO;AACLC,IAAAA,SAAS,EAAE,IADN;AAELC,IAAAA,KAAK,EAAEV,MAAM,CAAC,OAAD,EAAUI,qBAAV,EAAiCA,qBAAqB,CAACG,OAAvD,CAFR;AAGLI,IAAAA,IAAI,EAAEZ,MAAM,CAAC,MAAD,EAAS,EAAT;AAHP,GAAP;AAKD,CAND;;AAQAE,SAAS,CAAC,sCAAD,EAAyCW,MAAzC,CAAT,CAA0DC,GAA1D,CAA8D,SAA9D,EAAyE;AAAA,SACvE,oBAAC,gBAAD,EAAsBL,KAAK,EAA3B,CADuE;AAAA,CAAzE","sourcesContent":["import { number, radios } from '@storybook/addon-knobs'\nimport { storiesOf } from '@storybook/react'\nimport * as React from 'react'\nimport { LoadingIndicator, LoadingIndicatorColor, LoadingIndicatorProps } from './LoadingIndicator'\n\nconst loadingIndicatorColor: { [key: string]: LoadingIndicatorColor } = {\n  white: 'white',\n  black: 'black',\n  primary: 'primary',\n}\n\nconst knobs = (): LoadingIndicatorProps => {\n  return {\n    isVisible: true,\n    color: radios('Color', loadingIndicatorColor, loadingIndicatorColor.primary),\n    size: number('Size', 50),\n  }\n}\n\nstoriesOf('Design System/Atoms/LoadingIndicator', module).add('Default', () => (\n  <LoadingIndicator {...knobs()} />\n))\n"],"file":"LoadingIndicator.story.js"}