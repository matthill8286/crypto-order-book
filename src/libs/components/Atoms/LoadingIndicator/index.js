export * from "./LoadingIndicator";
//# sourceMappingURL=index.js.map