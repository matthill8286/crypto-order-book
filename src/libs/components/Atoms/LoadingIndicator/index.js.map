{"version":3,"sources":["../../../../src/components/Atoms/LoadingIndicator/index.ts"],"names":[],"mappings":"AAAA","sourcesContent":["export * from './LoadingIndicator'\n"],"file":"index.js"}