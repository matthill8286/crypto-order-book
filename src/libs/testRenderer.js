import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { saiyanTheme } from "./styles/sc-vars-saiyan";
export function renderWithTheme(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, tree)).toJSON();
}
export function renderWithThemeAndRouter(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, React.createElement(Router, null, tree))).toJSON();
}
export function renderWithRouter(tree) {
  return renderer.create(React.createElement(Router, null, tree)).toJSON();
}
export function mountWithTheme(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, tree));
}
export function mountWithThemeAndRouter(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, React.createElement(Router, null, tree)));
}
//# sourceMappingURL=testRenderer.js.map