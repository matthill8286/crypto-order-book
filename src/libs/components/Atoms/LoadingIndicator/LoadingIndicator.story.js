import { number, radios } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { LoadingIndicator } from "./LoadingIndicator";
var loadingIndicatorColor = {
  white: 'white',
  black: 'black',
  primary: 'primary'
};

var knobs = function knobs() {
  return {
    isVisible: true,
    color: radios('Color', loadingIndicatorColor, loadingIndicatorColor.primary),
    size: number('Size', 50)
  };
};

storiesOf('Design System/Atoms/LoadingIndicator', module).add('Default', function () {
  return React.createElement(LoadingIndicator, knobs());
});
//# sourceMappingURL=LoadingIndicator.story.js.map