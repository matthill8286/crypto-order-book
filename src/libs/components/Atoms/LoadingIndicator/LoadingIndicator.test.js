import * as React from 'react';
import { renderWithTheme } from "../../../testRenderer";
import { LoadingIndicator } from "./LoadingIndicator";
describe('LoadingIndicator', function () {
  it('renders saiyan theme', function () {
    var tree = renderWithTheme(React.createElement(LoadingIndicator, {
      isVisible: false
    }));
    expect(tree).toMatchSnapshot();
  });
});
//# sourceMappingURL=LoadingIndicator.test.js.map