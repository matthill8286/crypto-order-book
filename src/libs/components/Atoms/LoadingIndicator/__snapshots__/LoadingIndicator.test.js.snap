// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator renders saiyan theme 1`] = `
<div
  className="LoadingIndicator__StyledLoadingWrapper-sc-12we8y6-0 ibTRmW"
  color="primary"
  size={50}
>
  <div />
  <div />
  <div />
  <div />
</div>
`;
